fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexKeyValueToPartialRecordUtils.java

```diff
@@ -311,6 +324,7 @@ public void copy(@Nonnull Descriptors.Descriptor recordDescriptor, @Nonnull Mess
             String value = (String) keyTuple.get(keyTuple.size() - 1);
             Tuple groupingKey = Tuple.fromList(keyTuple.getItems().subList(0, keyTuple.size() - 2));
             buildPartialRecord(kv.getIndex().getRootExpression(), recordDescriptor, recordBuilder, fieldName, value, groupingKey);
+            populatePrimaryKey(primaryKeyExpression, recordDescriptor, recordBuilder, kv.getPrimaryKey());
```

Comment:
It feels like there should be tests somewhere that validate that:

1. The `IndexEntry`s that get produced from auto-complete have primary keys. (Possibly verifying that the primary key matches expectations, possibly just validating that calling `IndexEntry::getPrimaryKey()` doesn't throw an error and returns something non-`null`)
2. The partial records that get produced copy over the primary key from the entry. I'm not sure if we have tests that do the partial record construction, but if we do, making sure the primary key is set correctly sounds like a good check to add

Reply:
@alecgrieser the tests for 1 can be found from ```LuceneIndexTest``` now. It actually verifies more than just ```getPrimaryKey()``` is null. The tests for 2 can be found from ```LuceneDocumentFromRecordTest```. And the test  ```searchForAutoCompleteForGroupedRecord()``` in ```LuceneIndexTest``` also verified the partial record.